packages/live-share-inking/src/canvas/DryWetCanvas.ts

```diff
@@ -0,0 +1,269 @@
+                previousPoint = this._points[this._pendingPointsStartIndex - 1];
+                previousPointPressureAdjustedTip = getPressureAdjustedSize(tipHalfSize, previousPoint.pressure);
+            }
+
+            for (let i = this._pendingPointsStartIndex; i < this._points.length; i++) {
+                const p = this._points[i];
+
+                let pressureAdjustedTip = getPressureAdjustedSize(tipHalfSize, p.pressure);
+
+                const pathItem: IQuadPathItem = {
+                    endPoint: p,
+                    tipSize: pressureAdjustedTip
+                };
+
+                if (previousPoint !== undefined) {
+                    pathItem.quad = this.brush.tip === "ellipse"
+                        ? computeQuadBetweenTwoCircles(
+                            p,
+                            pressureAdjustedTip,
+                            previousPoint,
+                            previousPointPressureAdjustedTip)
+                        : computeQuadBetweenTwoRectangles(
+                            p,
+                            pressureAdjustedTip,
+                            pressureAdjustedTip,
+                            previousPoint,
+                            previousPointPressureAdjustedTip,
+                            previousPointPressureAdjustedTip);
+                }
+
+                result.push(pathItem);
+
+                previousPoint = p;
+                previousPointPressureAdjustedTip = pressureAdjustedTip;
+            }
+        }
+
+        return result;
+    }
+
+    private renderQuadPath(context: CanvasRenderingContext2D, path: IQuadPathItem[]) {
+        const cssColor = this.getBrushCssColor();
+
+        context.strokeStyle = cssColor;
+        context.fillStyle = cssColor;
+
+        context.beginPath();
+
+        for (let item of path) {
+            if (item.quad !== undefined) {
+                this.renderQuad(context, item.quad);
+            }
+
+            if (this.brush.tip === "ellipse") {
+                this.renderCircle(context, item.endPoint, item.tipSize);
+            }
+            else {
+                this.renderRectangle(
+                    context,
+                    item.endPoint,
+                    item.tipSize,
+                    item.tipSize);
+            }
+        }
+
+        context.fill();
+        context.closePath();
+    }
+
+    /**
+     * Defines the brush this canvas uses by default.
+     * @returns The brush this canvas should use by default.
+     */
+    protected getDefaultBrush(): IBrush {
+        return DefaultPenBrush;
+    }
+
+    /**
+     * Converts the current brush's color to a CSS color.
+     * @param color The color to convert.
+     * @returns A CSS color. 
+     */
+    protected getBrushCssColor(): string {
+        return toCssColor(this.brush.color, this.brush.type === "highlighter" ? 0.5 : 1);
+    }
+
+    /**
+     * Determines if this canvas renders strokes progressively. When rendering
```

Comment:
Leave note that this is for testing?

Reply:
This isn't for testing. It's two modes the canvas can work in. One is used by DryCanvas (non-progressive) the other by WetCanvas (progressive)